great_expectations/datasource/sqlalchemy_datasource.py

```diff
@@ -397,6 +399,9 @@ def get_batch(self, batch_kwargs, batch_parameters=None):
             )
         elif "table" in batch_kwargs:
             table = batch_kwargs["table"]
+            if batch_kwargs.get("use_quoted_name"):
+                table = quoted_name(table, quote=True)
+
```

Comment:
I think this *should* happen in the SqlalchemyDataset __init__ method, where table_name is being turned into the sqlalchemy Table object. Otherwise, we're ambiguous about whether table/table_name is a string or a SA clause. (See line 485 / 487 in sqlalchemy_dataset). That would require changing `get_init_kwargs` for SqlAlchemyBatchReference and ensuring the quoted_name kwarg is passed further in.

Because that whole API is being deprecated, that doesn't feel like an essential change to me.

Reply:
The trouble with this is that if there is a limit or offset, we are using the table_name to generate a query here, and it will be too late to use the quoted name in `SqlAlchemyDataset`. 

I might be misunderstanding your statement, but SA's `quoted_name` is a subclass of unicode/str, so I think of it more as a string-like object rather than a SA clause.

For simplicity's sake, I'm going to leave this as it is right now.